cmd/spire-server/cli/x509/mint.go

```diff
@@ -0,0 +1,185 @@
+	fs.Var(&c.dnsNames, "dns", "DNS name that will be included in SVID. Can be used more than once.")
+	fs.StringVar(&c.out, "out", "", "Directory to write output to instead of stdout")
+	return fs.Parse(args)
+}
+
+func (c *mintCommand) run() error {
+	if c.spiffeID == "" {
+		return errors.New("spiffeID must be specified")
+	}
+	key, err := c.generateKey()
+	if err != nil {
+		return fmt.Errorf("unable to generate key: %v", err)
+	}
+
+	csr, err := x509.CreateCertificateRequest(rand.Reader, &x509.CertificateRequest{}, key)
+	if err != nil {
+		return fmt.Errorf("unable to generate CSR: %v", err)
+	}
+
+	client, err := util.NewRegistrationClient(c.env.JoinPath(c.socketPath))
+	if err != nil {
+		return errors.New("cannot create registration client")
+	}
+
+	resp, err := client.MintX509SVID(context.Background(), &registration.MintX509SVIDRequest{
+		SpiffeId: c.spiffeID,
+		Csr:      csr,
+		Ttl:      int32(c.ttl / time.Second),
+		DnsNames: c.dnsNames,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to mint SVID: %v", err)
+	}
+
+	switch {
+	case len(resp.SvidChain) == 0:
+		return errors.New("server response missing SVID chain")
+	case len(resp.RootCas) == 0:
+		return errors.New("server response missing root CAs")
+	}
+
+	keyBytes, err := x509.MarshalPKCS8PrivateKey(key)
+	if err != nil {
+		return err
+	}
+
+	svidPEM := new(bytes.Buffer)
+	for _, certDER := range resp.SvidChain {
+		pem.Encode(svidPEM, &pem.Block{
+			Type:  "CERTIFICATE",
+			Bytes: certDER,
+		})
+	}
+
+	keyPEM := new(bytes.Buffer)
+	pem.Encode(keyPEM, &pem.Block{
+		Type:  "PRIVATE KEY",
+		Bytes: keyBytes,
+	})
+
+	bundlePEM := new(bytes.Buffer)
+	for _, rootCA := range resp.RootCas {
+		pem.Encode(bundlePEM, &pem.Block{
+			Type:  "CERTIFICATE",
+			Bytes: rootCA,
+		})
+	}
+
+	if c.out == "" {
+		if err := c.env.Printf("X509-SVID:\n%s\n", svidPEM.String()); err != nil {
+			return err
+		}
+		if err := c.env.Printf("Private key:\n%s\n", keyPEM.String()); err != nil {
+			return err
+		}
+		if err := c.env.Printf("Root CAs:\n%s\n", bundlePEM.String()); err != nil {
+			return err
+		}
+	} else {
+		svidPath := c.env.JoinPath(c.out, "svid.pem")
+		keyPath := c.env.JoinPath(c.out, "key.pem")
+		bundlePath := c.env.JoinPath(c.out, "bundle.pem")
+
+		if err := ioutil.WriteFile(svidPath, svidPEM.Bytes(), 0644); err != nil {
+			return fmt.Errorf("unable to write SVID: %v", err)
+		}
+		if err := c.env.Printf("X509-SVID written to %s.\n", svidPath); err != nil {
+			return err
+		}
+
+		if err := ioutil.WriteFile(keyPath, keyPEM.Bytes(), 0600); err != nil {
+			return fmt.Errorf("unable to write key: %v", err)
+		}
+		if err := c.env.Printf("Private key written to %s.\n", keyPath); err != nil {
+			return err
+		}
+
+		if err := ioutil.WriteFile(bundlePath, bundlePEM.Bytes(), 0644); err != nil {
+			return fmt.Errorf("unable to write bundle: %v", err)
+		}
+		if err := c.env.Printf("Root CAs written to %s.\n", bundlePath); err != nil {
```

Comment:
Wee nit: might be best to omit the trailing `.` on these messages so it's not misconstrued as being part of the filename

Reply:
Good nit. Fixed.